packages/components/src/internal/components/settings/DeleteProjectModal.tsx

```diff
@@ -0,0 +1,208 @@
+                placeholder="Enter comments (optional)"
+                value={comment}
+                rows={5}
+                cols={50}
+                onChange={onChangeComment}
+            />
+        </div>
+    );
+});
+
+export const BodyEmpty: FC<CommentAreaProps> = memo(({ comment, onChangeComment }) => {
+    return (
+        <Modal.Body>
+            <div className="delete-project-modal__text">
+                <p>This project will be permanently deleted. It contains no data.</p>
+            </div>
+
+            <CommentArea comment={comment} onChangeComment={onChangeComment} />
+        </Modal.Body>
+    );
+});
+
+interface BodyProps {
+    comment: string;
+    onChangeComment: ChangeEventHandler<HTMLTextAreaElement>;
+    summaries: Summary[];
+}
+export const Body: FC<BodyProps> = memo(({ summaries, comment, onChangeComment }) => {
+    return (
+        <Modal.Body>
+            <div className="delete-project-modal__text">
+                <p>This project and all of its data will be permanently deleted.</p>
+                <p>
+                    Before deleting this project, ensure there are no references to data (samples, sources or registry,
+                    assay data, etc.) in other projects.
+                </p>
+            </div>
+
+            <b> Project Data </b>
+
+            {summaries ? (
+                <table className="table table-responsive table-condensed delete-project-modal__table ">
+                    <tbody>
+                        {summaries.map(s => (
+                            <tr key={s.count + s.noun}>
+                                <td>{s.noun}</td>
+                                <td>{s.count}</td>
+                            </tr>
+                        ))}
+                    </tbody>
+                </table>
+            ) : (
+                <LoadingSpinner wrapperClassName="delete-project-modal__spinner" />
+            )}
+
+            <CommentArea comment={comment} onChangeComment={onChangeComment} />
+        </Modal.Body>
+    );
+});
+
+interface BodyDeletingProps {
+    totalCountFromSummaries: number;
+}
+export const BodyDeleting: FC<BodyDeletingProps> = memo(({ totalCountFromSummaries }) => {
+    const [toggle, seToggle] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+        setTimeout(() => seToggle(true), 1);
+    }, []);
+
+    return (
+        <Modal.Body>
+            <div className="deleting-project-modal-text">Please don't close this page until deleting is done.</div>
+            <Progress delay={0} estimate={totalCountFromSummaries * 15} toggle={toggle} />
+        </Modal.Body>
+    );
+});
+
+interface Props {
+    onCancel: () => void;
+    onError: (e: string) => void;
+    projectName: string;
+}
+
+export const DeleteProjectModal: FC<Props> = memo(props => {
+    const { projectName, onCancel, onError } = props;
+    const [summaries, setSummaries] = useState<Summary[]>(undefined);
```

Comment:
Consider using a loading state paradigm here and initializing summaries to `[]` so this component doesn't have to be continually defensive against it being undefined. Leads to less mistakes here and less error prone in the future for the next dev.